lib/rom/support/undefined.rb

```diff
@@ -0,0 +1,3 @@
+module ROM
+  Undefined = Object.new.freeze
+end
```

Comment:
ah, one small thing - could you move that to `rom.rb`? doesn't really deserve to be in a separate file (and requires are slow :))